ecommerce_integrations/unicommerce/invoice.py

```diff
@@ -570,3 +570,47 @@ def update_cancellation_status(so_data, so) -> bool:
 	from ecommerce_integrations.unicommerce.cancellation_and_returns import update_erpnext_order_items
 
 	update_erpnext_order_items(so_data, so)
+
+
+def on_submit(self, method=None):
```

Comment:
This hook will start triggering on invoices that aren't generated by unicommerce at all.


Make this code a bit more resilient? 


Consider all these use cases:
1. The invoice is not even generated by Unicommerce, should this hook even run? 
2. The invoice is generated via unicommerce but there's no picklist created for it, it's directly generated or manually generated.